app/code/Magento/Deploy/Package/Processor/PreProcessor/Css.php

```diff
@@ -180,11 +180,13 @@ private function buildMap($packagePath, $filePath, $fullPath)
      */
     private function collectFileMap($fileName)
     {
-        $result = isset($this->map[$fileName]) ? $this->map[$fileName] : [];
+        $result = [$this->map[$fileName] ?? []];
+
         foreach ($result as $path) {
-            $result = array_merge($result, $this->collectFileMap($path));
+            $result[] = $this->collectFileMap($path);
```

Comment:
An integration test is failing with provided changes: `\Magento\Deploy\DeployTest`
After the short investigation, we found that `$path` var can be the type of array and will break the check `$this->map[$fileName]`.